sonic_platform_base/sonic_xcvr/api/public/sff8636.py

```diff
@@ -403,3 +404,46 @@ def get_lpmode(self):
         # Since typically optics come up by default set to high power, in this case,
         # power_override not being set, function will return high power mode.
         return power_set and power_override
+
+    def get_power_class(self):
+        '''
+        Retrieves power class of the module.
+
+        Returns:
+            int: Power class of the module, -1 if it fails
+        '''
+        power_class_str = self.xcvr_eeprom.read(consts.POWER_CLASS_FIELD)
+        power_class = -1
+
+        if power_class_str is None:
+            return power_class
+        if not power_class_str.startswith(self.POWER_CLASS_PREFIX):
+            return power_class
+
+        prefix_len = len(self.POWER_CLASS_PREFIX)
+        power_class = int(power_class_str[prefix_len:prefix_len + 1])
+
+        return power_class
+
+    def set_high_power_class(self, enable):
+        '''
+        This function sets high power class for the module if needed.
+        It is only applicable for power class >= 5.
+
+        Args:
+            enable (bool): True to enable high power class, False to disable
+
+        Returns:
+            bool: True if the provision succeeds, False if it fails
+        '''
+        power_class = self.get_power_class()
+        ret = True
+
+        if power_class < 5:
+            return ret
+        elif power_class >= 8:
+            ret = self.xcvr_eeprom.write(consts.HIGH_POWER_CLASS_ENABLE_CLASS_8, enable)
+        else:  # Power class 5, 6, 7
+            ret = self.xcvr_eeprom.write(consts.HIGH_POWER_CLASS_ENABLE_CLASS_5_TO_7, enable)
+
```

Comment:
@longhuan-cisco you also need to make sure that the SW is in control for driving the power class. The way to do that is by doing a write of `1` to Byte 93 bit 0 i.e power override. Let do that in sff manager.

Reply:
@longhuan-cisco the point you are missing here is if you dont set the power override bit to 1 then LPmode must be de-asserted which is an extra thing to ensure. In SW control, where B93 bit 0 = 1, LPmode is don't care. Anyways, we should discuss this in your SFF manager PR

Reply:
Moved this discussion to sff_mgr PR at https://github.com/sonic-net/sonic-platform-daemons/pull/574#discussion_r1989878742